File: BankApp/src/app/Modules/Account.ts
import { AccountStatusType } from "./AccountStatusType";
import { Customer } from "./Customer";
import { SavingsInterestRate } from "./SavingsInterestRate";
import { TransactionLog } from "./TransactionLog";

export interface Account {
    accountId?: number | null;
    currentBalance?: number | null;
    interestSavingsRateId?: number | null;
    accountStatusTypeId?: number | null;
    customerId?: string | null;
    accountStatusType?: AccountStatusType | null;
    customer?: Customer | null;
    interestSavingsRate?: SavingsInterestRate | null;
    transactionLogs?: TransactionLog[];
}

File: BankApplicationAPI/BankApplicationAPI/Models/Account.cs
namespace BankApplicationAPI.Models;

public partial class Account
{
    public int? AccountId { get; set; }

    public decimal? CurrentBalance { get; set; }

    public byte? InterestSavingsRateId { get; set; }

    public byte? AccountStatusTypeId { get; set; }

    public string? CustomerId { get; set; }

    public virtual AccountStatusType? AccountStatusType { get; set; }

    public virtual Customer? Customer { get; set; }

    public virtual SavingsInterestRate? InterestSavingsRate { get; set; }

    public virtual ICollection<TransactionLog>? TransactionLogs { get; set; } = new List<TransactionLog>();
}
